packages/test-runner/src/fixtures.ts

```diff
@@ -33,7 +33,7 @@ export type TestInfo = {
   result: TestResult;
 };
 
-const registrations = new Map<string, FixtureRegistration>();
+const registrations = new Map<string, IncrementableArray<FixtureRegistration>>();
```

Comment:
With the change you are making, it seems like we should build explicit dependency tree of the fixture registrations. If one registration, fixture A will depend on B, in another it will depend on C and D.